sklearn/dummy.py

```diff
@@ -435,7 +435,7 @@ def fit(self, X, y, sample_weight=None):
 
             self.constant = check_array(self.constant,
                                         accept_sparse=['csr', 'csc', 'coo'],
-                                        ensure_2d=False)
+                                        ensure_2d=False, ensure_min_samples=0)
```

Comment:
BTW, I though I can make `_num_samples` return 1 for scalar / singleton arrays but I thought treating that edge-case that way might hide bugs. This is why I decided to do it that way. Let me know what you think.

singleton arrays are a weird beast. I think they are a design mistake in numpy but I did not want this PR to change the behavior of the `Dummy` models.